test/functionalTest/cases/1039_ngsiv2_date_support/dates_in_uri_param.test

```diff
@@ -0,0 +1,245 @@
+# Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Dates in URI param
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+
+--SHELL--
+
+#
+# 01. Create entity E24 with birthdate 2016-02-24T10:00:00.00Z
+# 02. Create entity E23 with birthdate 2016-02-23T10:00:00.00Z
+# 03. Create entity E22 with birthdate 2016-02-22T10:00:00.00Z
+# 04. GET entities with birthdate=2016-02-24T10:00:00.00Z, see E24
+# 05. GET entities with birthdate>2016-02-24T10:00:00.00Z, see nothing
+# 06. GET entities with birthdate<2016-02-25T10:00:00.00Z, see all three
+# 07. GET entities with birthdate>=2016-02-23T10:00:00.00Z, see E23 and E24
+# 08. GET entities with birthdate<=2016-02-23T10:00:00.00Z, see E22 and E23
+#
+
+echo "01. Create entity E24 with birthdate 2016-02-24T10:00:00.00Z"
+echo "============================================================"
+payload='{
+  "id": "E24",
+  "birthdate": {
+    "value": "2016-02-24T10:00:00.00Z",
+    "type": "date"
+  }
+}'
+orionCurl --url /v2/entities --payload "$payload" --json
+echo
+echo
+
+
+echo "02. Create entity E23 with birthdate 2016-02-23T10:00:00.00Z"
+echo "============================================================"
+payload='{
+  "id": "E23",
+  "birthdate": {
+    "value": "2016-02-23T10:00:00.00Z",
+    "type": "date"
+  }
+}'
+orionCurl --url /v2/entities --payload "$payload" --json
+echo
+echo
+
+
+echo "03. Create entity E22 with birthdate 2016-02-22T10:00:00.00Z"
+echo "============================================================"
+payload='{
+  "id": "E22",
+  "birthdate": {
+    "value": "2016-02-22T10:00:00.00Z",
+    "type": "date"
+  }
+}'
+orionCurl --url /v2/entities --payload "$payload" --json
+echo
+echo
+
+
+
+echo "04. GET entities with birthdate=2016-02-24T10:00:00.00Z, see E24"
+echo "================================================================"
+orionCurl --url '/v2/entities?q=birthdate==2016-02-24T10:00:00.00Z&options=keyValues' --json
```

Comment:
We should add a 04b, 05b, 06b, 07b and 08b without keyValues, to ensure that "type": "date" is correctly rendered for dates.

Maybe also options=values could make sense (or maybe it is excessive...).